elements/lisk-transaction-pool/src/transaction_pool.ts

```diff
@@ -109,9 +110,51 @@ export class TransactionPool {
 		return false;
 	}
 
-	public async addTransaction(tx: Transaction): Promise<boolean> {
-		console.log(tx);
-		return false;
+	public async addTransaction(incomingTx: Transaction): Promise<boolean> {
+		// Check for duplicate
+		if (this._allTransactions[incomingTx.id]) {
+			return false;
+		}
+		// Check for minimum entrance fee to the TxPool and if its low then reject the incoming tx
+		if (incomingTx.fee < this._minimumEntranceFee) {
+			return false;
+		}
+		// Check if incoming transaction fee is greater than the minimum fee in the TxPool if the TxPool is full
+		// TODO: Use feePriorityQueue to reject transaction with lowest fee when txPool is full
+
+		const incomingTxAddress = getAddressFromPublicKey(
+			incomingTx.senderPublicKey,
+		);
+
+		const txResponse = await this._applyFunction([incomingTx]);
+		const txResponseErrors = txResponse[0].errors;
+
+		// If applyTransaction fails for the transaction then throw error
+		if (
+			(txResponseErrors.length === 1 &&
+				txResponseErrors[0].dataPath === '.nonce' &&
+				txResponseErrors[0].actual &&
+				txResponseErrors[0].expected &&
+				txResponseErrors[0].actual > txResponseErrors[0].expected) ||
```

Comment:
Instead of having so many conditions you can use the new typescript `?` feature e.g. `txResponseErrors?.[0]?.actual`

Reply:
As discussed `txResponseErrors?.[0]?.actual` doesn't suppress ts error so keeping it like that